autoreload/autoreload.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+from lightning import Plugin
+import json
+import psutil
+import subprocess
+import threading
+import time
+import os
+
+plugin = Plugin()
+
+
+class ChildPlugin(object):
+
+    def __init__(self, path, plugin):
+        self.path = path
+        self.plugin = plugin
+        self.status = 'stopped'
+        self.proc = None
+        self.iolock = threading.Lock()
+        self.decoder = json.JSONDecoder()
+        self.manifest = None
+        self.init = None
+        self.reader = None
+
+    def watch(self):
+        last = os.path.getmtime(self.path)
+        while True:
+            time.sleep(1)
+            now = os.path.getmtime(self.path)
+            if last != now:
+                print("Detected a change in the child plugin, restarting...")
+                last = now
+                self.restart()
+
+    def handle_init(self, request):
+        """Lightningd has sent us its first init message, clean and forward.
+        """
+        raw = {
+            'jsonrpc': '2.0',
+            'method': request.method,
+            'params': request.params,
+            'id': request.id,
+        }
+
+        self.init = raw
+        del self.init['params']['plugin']
```

Comment:
You need to check of key `'plugin'` and `'request'`. Otherwise it didnt work (for me):

```Python
        if 'plugin' in self.init['params']:
            del self.init['params']['plugin']
        if 'request' in self.init['params']:
            del self.init['params']['request']
```

Note: `my self.init['params']` looked like this:
```
{'options': {'summary-currency': 'USD', 'summary-currency-prefix': 'USD $', 'autoreload-plugin': '/home/will/projects/lightning.plugins.git/summary/summary.py'}, 'configuration': {'lightning-dir': '/home/will/.lightning', 'rpc-file': 'lightning-rpc'}
```

Reply:
Right, I was working with my parameter binding patchset already applied, which behaves a bit differently. Made it conditional.